OBSDOCS-205 - CLF audit log filtering

libander opened the pull request:
<!--- PR title format: [GH#<gh-issue-id>][BZ#<bz-issue-id>][OCPBUGS#<jira-issue-id>][OSDOCS#<jira-issue-id>]: <short-description-of-the-pr> --->

<!--- If your changes apply to the latest release and/or in-development version of OpenShift, open your PR against the `main` branch.

* For more details about the information requested in this template, see:
  https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/create_or_edit_content.adoc#submit-PR --->

Version(s): 4.12+
<!--- Specify the version or versions of OpenShift your PR applies to. -->

Issue: [OBSDOCS-205](https://issues.redhat.com//browse/OBSDOCS-205)
<!--- Add a link to the Bugzilla, Jira, or GitHub issue, if applicable. --->

Link to docs preview: https://70608--ocpdocs-pr.netlify.app/openshift-enterprise/latest/logging/log_collection_forwarding/configuring-log-forwarding#logging_audit_filtering_configuring-log-forwarding
<!--- Add direct link(s) to the exact page(s) with updated content from the preview build. --->

QE review: @anpingli / @kabirbhartiRH / @QiaolingTang 

- [x] QE has approved this change.
<!--- QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

Additional information:
SME review: @alanconway / @jcantrill 
<!--- Optional: Include additional context or expand the description here.--->

<!--- After you open your PR, ask for review from the OpenShift docs team:
  For community authors: Tag @openshift/team-documentation in a GitHub comment.--->

libander commented:
/label peer-review-needed

QiaolingTang commented on modules/logging-audit-log-filtering.adoc:
```diff
@@ -0,0 +1,118 @@
+* None: The event is dropped.
+* Metadata: Only the audit metadata is included, request and response bodies are removed.
+* Request: Audit metadata and the request body are included, the response body is removed.
+* RequestResponse: All data is included: metadata, request body and response body. The response body can be very large. For example, `oc list -A pods` generates a response body containing the YAML description of every pod in the cluster.
```
Seems `oc list` is not a valid command, I got:

```
$ oc list
error: unknown command "list" for "oc"

Did you mean this?
	get
	wait
```

libander replied:
@QiaolingTang - fixed, thank you!

snarayan-redhat commented:
@libander I would suggest submitting this for peer review once you get a QE approval.

JoeAldinger commented:
@libander do you mind adding the assembly file so the preview works, please?

libander commented:
/retest

libander commented:
/label peer-review-needed

QiaolingTang commented:
LGTM.

snarayan-redhat reviewed (commented):
Good effort. I have left few suggestions which might want to consider.

snarayan-redhat commented:
For now, I am not adding a `peer-review-done` label since there are some comments that need to be addressed. You can request for a review once ready by adding `peer-review-needed` label again.

jcantrill reviewed (commented):
/lgtm

libander commented on modules/logging-audit-log-filtering.adoc:
```diff
@@ -0,0 +1,118 @@
+// Module included in the following assemblies:
+//
+//
```
Good catch, thanks!

libander commented:
/label peer-review-needed

dfitzmau reviewed (commented):
Hi @libander . Nice work on the contnet. I added some minor comments but the YAML file might indicate a bigger issue for customers that copy then example verbatim. I see some red text highlighting in the preview that might indicate an issue.

libander commented:
> Hi @libander . Nice work on the contnet. I added some minor comments but the YAML file might indicate a bigger issue for customers that copy then example verbatim. I see some red text highlighting in the preview that might indicate an issue.

Thanks! Added admonitions to those fields, and corrected the admonition. That example is not intended to be copied as it isn't a recommended policy. :)

libander commented:
/label peer-review-needed

libander commented:
/label merge-review-needed

libander commented:
/remove-label peer-review-needed

mburke5678 commented on modules/logging-audit-log-filtering.adoc:
```diff
@@ -0,0 +1,117 @@
+
+[NOTE]
+====
+The example provided is intended to illustrate of the range of rules possible in an audit policy and is not a recommended configuration.
```
```suggestion
The example provided is intended to illustrate the range of rules possible in an audit policy and is not a recommended configuration.
```

mburke5678 commented on modules/logging-audit-log-filtering.adoc:
```diff
@@ -0,0 +1,117 @@
+  pipelines:
+    - name: my-pipeline
+      inputRefs: #<1>
+      filterRefs: #<2>
```
Would it make sense to add some examples to the `inputRefs` and `filterRefs` parameters, especially where the code example is purely illustrative. It took me several minutes to figure it out (assuming I did):

```
  pipelines:
    - name: my-pipeline
      inputRefs: audit #<1>
      filterRefs: my-policy #<2>
```

mburke5678 commented:
@libander I have a couple of nits I was hoping you could look at before we merge.

mburke5678 commented:
/cherrypick enterprise-4.12

mburke5678 commented:
/cherrypick enterprise-4.13

mburke5678 commented:
/cherrypick enterprise-4.14

mburke5678 commented:
/cherrypick enterprise-4.15
